packages/nx-monorepo/src/nx-monorepo.ts

```diff
@@ -776,8 +773,13 @@ export class NxMonorepoProject extends TypeScriptProject {
       // @ts-ignore - `file` is private
       subProjectPackages.find((pkg) => (pkg.file as JsonFile).changed === true)
     ) {
-      // @ts-ignore - `installDependencies` is private
-      this.package.installDependencies();
+      try {
+        // @ts-ignore - `installDependencies` is private
+        this.package.installDependencies();
+      } finally {
+        // @ts-ignore - `resolveDepsAndWritePackageJson` is private
+        this.package.resolveDepsAndWritePackageJson();
+      }
```

Comment:
Missing a `catch` here?

This logic will always run `installDependencies` and `resolveDepsAndWritePackageJson`, even if `installDependencies` fails, but the closure will still fail afterwards if `installDependencies` fails.

I think you want an empty catch here.

Reply:
It's actually correct as it - I want to throw but need to make sure the package.json is written out so it doesn't end up in a corrupt state.